backend/dataall/core/environment/services/environment_service.py

```diff
@@ -120,120 +150,189 @@ def validate_permissions(session, uri, g_permissions, group):
+                subnet_ids=data.get('subnetIds', []),
+            )
+
+        return cdk_role_name
+
     @staticmethod
     @TenantPolicyService.has_tenant_permission(MANAGE_ENVIRONMENTS)
     @ResourcePolicyService.has_resource_permission(LINK_ENVIRONMENT)
-    def create_environment(session, uri, data=None):
+    def create_environment(uri, data=None):
         context = get_context()
-        EnvironmentRequestValidationService.validate_creation_params(data, uri, session)
-        organization = OrganizationRepository.get_organization_by_uri(session, uri)
-        env = Environment(
-            organizationUri=data.get('organizationUri'),
-            label=data.get('label', 'Unnamed'),
-            tags=data.get('tags', []),
-            owner=context.username,
-            description=data.get('description', ''),
-            environmentType=data.get('type', EnvironmentType.Data.value),
-            AwsAccountId=data.get('AwsAccountId'),
-            region=data.get('region'),
-            SamlGroupName=data['SamlGroupName'],
-            validated=False,
-            isOrganizationDefaultEnvironment=False,
-            EnvironmentDefaultIAMRoleName=data.get('EnvironmentDefaultIAMRoleArn', 'unknown').split('/')[-1],
-            EnvironmentDefaultIAMRoleArn=data.get('EnvironmentDefaultIAMRoleArn', 'unknown'),
-            CDKRoleArn=f"arn:aws:iam::{data.get('AwsAccountId')}:role/{data['cdk_role_name']}",
-            resourcePrefix=data.get('resourcePrefix'),
-        )
+        with context.db_engine.scoped_session() as session:
+            EnvironmentRequestValidationService.validate_creation_params(data, uri, session)
+            cdk_role_name = EnvironmentService.check_cdk_resources(data.get('AwsAccountId'), data.get('region'), data)
+            organization = OrganizationRepository.get_organization_by_uri(session, uri)
+            env = Environment(
+                organizationUri=data.get('organizationUri'),
+                label=data.get('label', 'Unnamed'),
+                tags=data.get('tags', []),
+                owner=context.username,
+                description=data.get('description', ''),
+                environmentType=data.get('type', EnvironmentType.Data.value),
+                AwsAccountId=data.get('AwsAccountId'),
+                region=data.get('region'),
+                SamlGroupName=data['SamlGroupName'],
+                validated=False,
+                isOrganizationDefaultEnvironment=False,
+                EnvironmentDefaultIAMRoleName=data.get('EnvironmentDefaultIAMRoleArn', 'unknown').split('/')[-1],
+                EnvironmentDefaultIAMRoleArn=data.get('EnvironmentDefaultIAMRoleArn', 'unknown'),
+                CDKRoleArn=f"arn:aws:iam::{data.get('AwsAccountId')}:role/{cdk_role_name}",
+                resourcePrefix=data.get('resourcePrefix'),
+            )
 
-        session.add(env)
-        session.commit()
+            env.userRoleInEnvironment = EnvironmentPermission.Owner.value
```

Comment:
do we need this line? If we resolve user role as a subquery using `resovle_user_role()`?

Reply:
Yes, I guess we can drop it